### PR TITLE
[API-289] [API-290] [API-291] SQL documentation, code samples

### PR DESCRIPTION
**Pull request by srknzl**
merge https://github.com/hazelcast/hazelcast-nodejs-client/pull/847 first

### CODE REVIEW

**Comment by srknzl** [comment]
verify

**Comment by srknzl** [comment]
verify

**Comment by srknzl** [comment]
verify

**Comment by srknzl** [comment]
verify

**Comment by mdumandag** [comment]
verify

**Comment by srknzl** [comment]
verify

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2210]:
```suggestion
for await (const row of result) {
```

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2215]: We have a separate section for casting, I think it will be better if we move this to there

File: .eslintrc
Comment (srknzl) [inline .eslintrc:103]: needed for code samples, so moved from test/eslintrc to root level eslintrc

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2263]:
```suggestion
#### Fields
```

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2340]: Let's remove the Java types and mention of Java below

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2378]: unnecessary newlines

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2382]: unnecessary newline

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2237]: ... other types that are comparable ...

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2284]: I think there should be a new line before this bulleted list item (to render it properly).

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2283]: ... fields of a key or value.

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2286]: ... objects, the fields written with ...

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2383]: Note that the default number type in Node.js client is double. This ...

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2340]: Data Types

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2369]: How to Cast

File: DOCUMENTATION.md
Comment (Serdaro) [inline DOCUMENTATION.md:2380]: Important Notes About Comparison and Casting

**Review by Serdaro — COMMENTED** [review]
Minor comments

File: code_samples/sql-data-types.js
Comment (srknzl) [reply code_samples/sql-data-types.js:132]: `Error occurred: HazelcastSqlException [Error]: From line 1, column 15 to line 1, column 24: Object 'decimalMap' not found`
 
thrown while executing sql, if map is empty

File: code_samples/sql-data-types.js
Comment (srknzl) [reply code_samples/sql-data-types.js:125]: It is impossible to add decimal using nodejs client, so I am not sure what to write there; because writing DECIMAL would be wrong as it wont print DECIMAL

File: code_samples/sql-data-types.js
Comment (srknzl) [reply code_samples/sql-data-types.js:42]: I think showing all data types, especially DECIMAL, and DATE/TIME related ones are important. REAL is also important since we are showing a CAST is necessary there. Another option is to just show a couple; but then the user will have some difficulty using sql.

Comment (srknzl) [reply code_samples/sql-data-types.js:42]: Because casting is necessary for some of the types.

File: DOCUMENTATION.md
Comment (srknzl) [reply DOCUMENTATION.md:2215]: move what?

Comment (srknzl) [reply DOCUMENTATION.md:2215]: I think now I got it, move this section to 8.7.3

File: code_samples/sql-data-types.js
Comment (mdumandag) [reply code_samples/sql-data-types.js:42]: I believe if the type is not easily usable from Node.js client, we can omit it. My reasoning is that, such types will require Java client access, and the users who know how to use both clients can understand what to do looking at the documentation. Besides, when a user tries to run this code sample, he will see errors instead of outputs and maybe think something is broken

Comment (mdumandag) [reply code_samples/sql-data-types.js:42]: I would just put string, portable and number examples and that would probably be enough

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2239]: unnecessary newline

File: code_samples/sql-data-types.js
Comment (srknzl) [reply code_samples/sql-data-types.js:42]: is current integers example ok? or we should have single integer type?

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2334]:
```suggestion
SQL `DATE` type is sent and received as a string with the `yyyy-mm-dd` format.
```

File: code_samples/sql-data-types.js
Comment (srknzl) [reply code_samples/sql-data-types.js:42]: hm ok now I saw your note in parantheses, sorry

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2338]:
```suggestion
SQL `TIME` type is sent and received as string  with the `HH:mm:ss.SSS` format where `SSS` represents nanoseconds and can
```

Comment (mdumandag) [reply DOCUMENTATION.md:2338]: similar thing can be applied to other date types

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2369]:
```suggestion
as default number type, to compare with an integer based column you can use Long objects or casting.
```

File: code_samples/sql-data-types.js
Comment (mdumandag) [reply code_samples/sql-data-types.js:42]: I would choose the one you need a cast and explain the reasoning

**Comment by srknzl** [comment]
@mdumandag can I get a final review? I think we are almost there <3

File: code_samples/sql-data-types.js
Comment (srknzl) [inline code_samples/sql-data-types.js:62]: going to change `for these types` to `for BIGINT`

File: DOCUMENTATION.md
Comment (mdumandag) [inline DOCUMENTATION.md:2333]:
```suggestion
`DECIMAL` SQL type is sent and received as strings.
```

**Review by mdumandag — APPROVED** [review]
approving with some minor comments left

**Pull request merged**
